### ISSUE TITLE
Panic in ConnManager when starting testnet

### ISSUE THREAD

**Issue by carryforward** [btcsuite/btcd#849]
I start btcd without a config file.  I launch it with this command:

btcd --testnet

I am running commit 089611a61b01c45d21163707d3e1d4bc80b28f39

After a short, but random amount of time (at most 5 minutes, often sooner) btcd panics.

I get this panic:

```
panic: runtime error: index out of range

goroutine 4932 [running]:
panic(0x9a6540, 0xc4200100d0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
main.addrStringToNetAddr(0xc42549eec0, 0x1c, 0x1c, 0x5, 0x1, 0xc42595bd70)
        /home/factom/go/src/github.com/btcsuite/btcd/server.go:2526 +0x1a4
main.newServer.func4(0xc42663b900, 0x19, 0x19, 0x0)
        /home/factom/go/src/github.com/btcsuite/btcd/server.go:2443 +0x350
github.com/btcsuite/btcd/connmgr.(*ConnManager).NewConnReq(0xc420488900)
        /home/factom/go/src/github.com/btcsuite/btcd/connmgr/connmanager.go:288 +0xc6
created by github.com/btcsuite/btcd/connmgr.(*ConnManager).handleFailedConn
        /home/factom/go/src/github.com/btcsuite/btcd/connmgr/connmanager.go:207 +0x442
```

the full log is attached: [log.txt](https://github.com/btcsuite/btcd/files/586985/log.txt)
again with trace: [log2.txt](https://github.com/btcsuite/btcd/files/586988/log2.txt)

This line might be the one causing the panic: https://github.com/btcsuite/btcd/blob/089611a61b01c45d21163707d3e1d4bc80b28f39/server.go#L2526

it might have been this commit that caused it:

https://github.com/btcsuite/btcd/commit/e8f63bc29550705268b533032ccc2ea24f8c86ba#diff-34c6b408d72845d076d47126c29948d1R2526

@Roasbeef 

current workaround is to disable dns seeds.
> nslookup testnet-seed.bitcoin.schildbach.de
to get a testnet address
> btcd --testnet --nodnsseed -a <address from above>

**Comment by davecgh** [comment]
Thanks for the report.  You're correct about that being issue.  I'll push a fix shortly.

**Comment by davecgh** [comment]
Fixed by #850.

**Comment by carryforward** [comment]
That was fast.  Thank you.  The fix worked on my setup.
